frontend/src/metabase/search/components/filters/CreatedByFilter/CreatedByContent.tsx

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable react/prop-types */
+import { useState } from "react";
+import { isEqual } from "underscore";
+import { t } from "ttag";
+import { useUserListQuery } from "metabase/common/hooks/use-user-list-query";
+import type { UserListResult } from "metabase-types/api";
+import { Center, Loader, Text, TextInput } from "metabase/ui";
+import {
+  CreatedByContainer,
+  CreatedByContentContainer,
+} from "metabase/search/components/filters/CreatedByFilter/CreatedByContent.styled";
+import type { SearchSidebarFilterComponent } from "metabase/search/types";
+import { UserListElement } from "metabase/search/components/filters/CreatedByFilter/UserListElement";
+
+export const CreatedByContent: SearchSidebarFilterComponent<"created_by">["ContentComponent"] =
+  ({ value, onChange }) => {
+    const { data: users = [], isLoading } = useUserListQuery();
+    const [userFilter, setUserFilter] = useState("");
+
+    const filteredUsers = users.filter(user =>
+      user.common_name.toLowerCase().includes(userFilter.toLowerCase()),
+    );
+
+    const onUserSelect = (user: UserListResult) => {
+      if (value && value.length > 0 && isEqual(value[0], String(user.id))) {
+        onChange([]);
+      } else {
+        onChange([String(user.id)]);
+      }
```

Comment:
1. Why `CreatedByFilterProps = string[]` and not `CreatedByFilterProps = UserId[]` ?
2. Why is it an array if it only ever has 1 or 0 elements?

Reply:
1. Good call
2. I was thinking about this for a while. There's a two reasons:
    - Our type filter logic supports multiple elements, so the logic for supporting arrays is already there, and I didn't want to go out of my way to make more changes than necessary to allow for a single element.
    - If we ever find that there is demand for filtering by multiple users (maybe someone wants to know about dashboards created by multiple people or something) then all we have to do is change this function. So, in essence, future proofing at a very low cost

Reply:
> So, in essence, future proofing at a very low cost

- YAGNI
- it's confusing why it's there
- if it was future-proof `value[0]` would never appear in this PR, but there are 3 occurrences in 2 files

Reply:
I don't think it's too confusing considering it's three occurrences. On the `YAGNI` thing, I wholeheartedly disagree, but ok. I'll change it if you think it's the better way to go.